cdap-api/src/main/java/co/cask/cdap/api/artifact/ArtifactId.java

```diff
@@ -49,11 +49,7 @@ public ArtifactScope getScope() {
 
   @Override
   public String toString() {
-    return "ArtifactId{" +
-      "name='" + name + '\'' +
-      ", version=" + version +
-      ", scope='" + scope + '\'' +
-      '}';
+    return String.format("%s-%s-%s", scope, name, version);
```

Comment:
The generation of file name shouldn't be part of `ArtifactId`. This class should be pure data. You can have helper method in app-fabric to unify generation of file name from artifactId.